File: data/src/notFoss/java/org/cryptomator/data/cloud/googledrive/RootGoogleDriveFolder.java
package org.cryptomator.data.cloud.googledrive;

import org.cryptomator.domain.Cloud;
import org.cryptomator.domain.GoogleDriveCloud;

public class RootGoogleDriveFolder extends GoogleDriveFolder {

	private final GoogleDriveCloud cloud;

	public RootGoogleDriveFolder(GoogleDriveCloud cloud) {
		super(null, "", "", "root");
		this.cloud = cloud;
	}

	@Override
	public GoogleDriveCloud getCloud() {
		return cloud;
	}

	@Override
	public GoogleDriveFolder withCloud(Cloud cloud) {
		return new RootGoogleDriveFolder((GoogleDriveCloud) cloud);
	}
}

File: data/src/notFoss/java/org/cryptomator/data/cloud/googledrive/RootGoogleDriveFolder.kt
package org.cryptomator.data.cloud.googledrive

import org.cryptomator.domain.Cloud
import org.cryptomator.domain.GoogleDriveCloud

class RootGoogleDriveFolder(override val cloud: GoogleDriveCloud) : GoogleDriveFolder(null, "", "", "root") {

	override fun withCloud(cloud: Cloud?): GoogleDriveFolder {
		return RootGoogleDriveFolder(cloud as GoogleDriveCloud)
	}
}
